File: DragonsGenerator.WEB/src/app/core/models/Equipments/equipment.ts
import { Cost } from './cost';
import { EquipmentData } from './equipment-data';
import { EquipmentSubtype, EquipmentType } from './equipment-enums';

export interface Equipment {
  id: string;
  name: string;
  type: EquipmentType;
  subtype: EquipmentSubtype | null;
  cost: Cost;
  wKg: number | null;
  data: EquipmentData | null;
}

File: DragonsGenerator.API/Models/Equipment.cs
using System.Text.Json;

namespace DragonsGenerator.API.Models;

public record Equipment(
    string Id,
    string Name,
    string Type,
    Cost Cost,
    JsonElement? Data,
    string? Subtype = null,
    double? WKg = null
);

public record Cost(int? V, string U);